packages/lisk-transactions/src/utils/validation/schema.js

```diff
@@ -0,0 +1,282 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+export const baseTransaction = {
+	$id: 'lisk/base-transaction',
+	type: 'object',
+	required: [
+		'id',
+		'type',
+		'amount',
+		'fee',
+		'senderPublicKey',
+		'recipientId',
+		'timestamp',
+		'asset',
+		'signature',
+	],
+	properties: {
+		id: {
+			type: 'string',
+			format: 'number',
+			minLength: 1,
+			maxLength: 20,
+		},
+		amount: {
+			type: 'string',
+			format: 'number',
+		},
+		fee: {
+			type: 'string',
+			format: 'number',
+		},
+		type: {
+			type: 'integer',
+			minimum: 0,
+			maximum: 7,
+		},
+		timestamp: {
+			type: 'integer',
+		},
+		senderId: {
+			type: 'string',
+			format: 'address',
+		},
+		senderPublicKey: {
+			type: 'string',
+			format: 'publicKey',
+		},
+		senderSecondPublicKey: {
+			type: 'string',
+			format: 'publicKey',
+		},
+		recipientId: {
+			type: 'string',
+		},
+		recipientPublicKey: {
+			type: ['string', 'null'],
+			format: 'publicKey',
+		},
+		signature: {
+			type: 'string',
+			format: 'signature',
+		},
+		signSignature: {
+			type: 'string',
+			format: 'signature',
+		},
+		signatures: {
+			type: 'array',
+			uniqueItems: true,
+			items: {
+				type: 'string',
+				format: 'signature',
+			},
+		},
+		asset: {
+			type: 'object',
+		},
+	},
+};
+
+export const transferTransaction = {
+	$merge: {
+		source: { $ref: 'lisk/base-transaction' },
+		with: {
+			properties: {
+				amount: {
+					type: 'string',
+					format: 'number',
+					minLength: 1,
+				},
+				recipientId: {
+					format: 'address',
+				},
+				asset: {
+					type: 'object',
+					properties: {
+						data: {
+							type: 'string',
+							maxLength: 32,
+						},
+					},
+				},
+			},
+		},
+	},
+};
+
+export const signatureTransaction = {
+	$merge: {
+		source: { $ref: 'lisk/base-transaction' },
+		with: {
+			properties: {
+				asset: {
+					type: 'object',
+					required: ['signature'],
+					properties: {
+						signature: {
+							type: 'object',
+							required: ['publicKey'],
+							properties: {
+								publicKey: {
+									type: 'string',
+									format: 'publicKey',
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+};
+
+export const delegateTransaction = {
+	$merge: {
+		source: { $ref: 'lisk/base-transaction' },
+		with: {
+			properties: {
+				asset: {
+					type: 'object',
+					required: ['delegate'],
+					properties: {
+						delegate: {
+							type: 'object',
+							required: ['username'],
+							properties: {
+								username: {
+									type: 'string',
+									maxLength: 10,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+};
+
+export const voteTransaction = {
+	$merge: {
+		source: { $ref: 'lisk/base-transaction' },
+		with: {
+			properties: {
+				asset: {
+					type: 'object',
+					required: ['votes'],
+					properties: {
+						votes: {
+							type: 'array',
+							uniqueItems: true,
```

Comment:
This won't include voting and unvoting for the same delegate in a vote transaction. (Is that scenario even invalid?)